src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1715,6 +1719,73 @@ private void SetPeriodicEvaluation(WorkspaceModel ws)
+
+            // Get the dummy nodes in the current workspace. 
+            var dummyNodes = currentWorkspace.Nodes.OfType<DummyNode>();
+
+            foreach (DummyNode dn in dummyNodes)
+            {
+                dummyNodeJSON = dn.OriginalNodeContent as JObject;
+
+                if (dummyNodeJSON != null)
+                {
+                    // Deserializing the dummy node and verifying if it is resolved by the downloaded package
+                    NodeModel resolvedNode = dn.GetNodeModelForDummyNode(
+                                                               dummyNodeJSON.ToString(),
+                                                               LibraryServices,
+                                                               NodeFactory,
+                                                               IsTestMode,
+                                                               CustomNodeManager);
+
+                    // If the resolved node is also a dummy node, then skip it else replace the dummy node with the resolved version of the node. 
+                    if (!(resolvedNode is DummyNode))
+                    {
+                        // Disable graph runs temporarily while the dummy node is replaced with the resolved version of that node.
+                        EngineController.DisableRun = true;
+                        currentWorkspace.RemoveAndDisposeNode(dn);
+                        currentWorkspace.AddAndRegisterNode(resolvedNode, false);
+
+                        // Adding back the connectors for the resolved node. 
+                        List<ConnectorModel> connectors = dn.AllConnectors.ToList();
+                        foreach (var connectorModel in connectors)
+                        {
+                            var startNode = connectorModel.Start.Owner;
+                            var endNode = connectorModel.End.Owner;
+
+                            if (startNode is DummyNode && startNode.GUID == resolvedNode.GUID)
+                            {
+                                startNode = resolvedNode;
+                            }
+                            if (endNode is DummyNode && endNode.GUID == resolvedNode.GUID)
+                            {
+                                endNode = resolvedNode;
+                            }
+
+                            connectorModel.Delete();
+                            ConnectorModel.Make(startNode, endNode, connectorModel.Start.Index, connectorModel.End.Index, connectorModel.GUID);
+                        }
+                        EngineController.DisableRun = false ;
+                    }
+                }
+            }
+
+            // Once all the dummy nodes are reloaded, the DummyNodesReloaded event is invoked and
+            // the Dependency table is regenerated in the WorkspaceDependencyView extension. 
+            currentWorkspace.OnDummyNodesReloaded();
```

Comment:
@reddyashish I think you should set the hasUnsavedChanges = false here manually and add a test for this - @aparajit-pratap what do you think?

Reply:
Setting hasUnsavedChanges=false and the workspace changes will be saved automatically as there are no changes to the serialized file. Will add this to the test as well.